crates/forge_repo/src/forge_repo.rs

```diff
@@ -487,6 +493,45 @@ impl<F: Send + Sync> forge_domain::WorkspaceRepository for ForgeRepo<F> {
     }
 }
 
+#[async_trait::async_trait]
+impl<F: EnvironmentInfra + Send + Sync> forge_domain::WorkspaceSyncRepository for ForgeRepo<F> {
```

Comment:
don't need a separate repo. The sync concern can be handled right inside the workspace repo implementation.